web/ui/src/Services/ServiceDetailsController.js

```diff
@@ -3,1076 +3,1107 @@
+    let $notification, serviceHealth, $q, resourcesFactory, utils;
+
     controlplane.controller("ServiceDetailsController",
-    ["$scope", "$q", "$routeParams", "$location", "resourcesFactory",
-    "authService", "$modalService", "$translate", "$notification",
-    "$timeout", "servicesFactory", "miscUtils", "hostsFactory",
-    "poolsFactory", "CCUIState", "$cookies", "areUIReady", "LogSearch",
-    function($scope, $q, $routeParams, $location, resourcesFactory,
-    authService, $modalService, $translate, $notification,
-    $timeout, servicesFactory, utils, hostsFactory,
-    poolsFactory, CCUIState, $cookies, areUIReady, LogSearch){
-
-        // Ensure logged in
-        authService.checkLogin($scope);
-        $scope.resourcesFactory = resourcesFactory;
-        $scope.hostsFactory = hostsFactory;
-
-        $scope.defaultHostAlias = $location.host();
-        if(utils.needsHostAlias($location.host())){
-            resourcesFactory.getHostAlias().success(function(data) {
-                $scope.defaultHostAlias = data.hostalias;
-            });
-        }
-
-        //add Public Endpoint data
-        $scope.publicEndpoints = { add: {} };
-
-        //add service endpoint data
-        $scope.exportedServiceEndpoints = { };
-
-        $scope.click_pool = function(id) {
-            resourcesFactory.routeToPool(id);
-        };
-
-        $scope.click_host = function(id) {
-            resourcesFactory.routeToHost(id);
-        };
-
-
-        $scope.modalAddPublicEndpoint = function() {
-            areUIReady.lock();
-            $scope.protocols = [];
-            $scope.protocols.push({ Label: "HTTPS", UseTLS: true, Protocol: "https" });
-            $scope.protocols.push({ Label: "HTTP", UseTLS: false, Protocol: "http" });
-            $scope.protocols.push({ Label: "Other, secure (TLS)", UseTLS: true, Protocol: "" });
-            $scope.protocols.push({ Label: "Other, non-secure", UseTLS: false, Protocol: "" });            
-            
-            // default public endpoint options
-            $scope.publicEndpoints.add = {
-                type: "port",
-                app_ep: $scope.exportedServiceEndpoints.data[0],
-                name: "",
-                host: $scope.defaultHostAlias,
-                port: "",
-                protocol: $scope.protocols[0],
-            };
-
-            // returns an error string if newPublicEndpoint's vhost is invalid
-            var validateVHost = function(newPublicEndpoint){
-                var name = newPublicEndpoint.name;
-
-                // if no port
-                if(!name || !name.length){
-                    return "Missing Name";
+        ["$scope", "$q", "$routeParams", "$location", "resourcesFactory",
+            "authService", "$modalService", "$translate", "$notification",
+            "$timeout", "miscUtils", "hostsFactory", "$serviceHealth", "Service",
+            "poolsFactory", "CCUIState", "$cookies", "areUIReady", "LogSearch",
```

Comment:
I'd like to eventually remove the servicesFactory but I don't know of plans to remove poolsFactory or hostsFactory.